common/views/components/WobblyEdge/WobblyEdge.js

```diff
@@ -1,18 +1,93 @@
 // @flow
+import React from 'react';
+import prefixedPropertyStyleObject from '../../../utils/prefixed-property-style-object';
+
+function randomIntFromInterval(min: number, max: number): number {
+  return Math.floor(Math.random() * (max - min + 1) + min);
+}
+
 type Props = {|
   background: string,
   intensity?: number,
   points?: number,
-  isValley?: boolean
+  isValley?: boolean,
+  isStatic?: boolean,
+  extraClasses?: string
+|}
+
+type State = {|
+  isActive: boolean,
+  styleObject: {}
 |}
 
-const WobblyEdge = ({ intensity, points, background, isValley }: Props) => (
-  <div
-    className={`wobbly-edge wobbly-edge--${background} js-wobbly-edge`}
-    data-max-intensity={intensity}
-    data-number-of-points={points}
-    data-is-valley={isValley}>
-  </div>
-);
+class WobblyEdge extends React.Component<Props, State> {
+  timer: any;
+  intensity: number;
+  points: number;
+
+  constructor(props: Props) {
+    super(props);
+    this.intensity = props.intensity || 50;
+    this.points = props.points || 5;
+    this.timer = null;
+    this.state = {
+      isActive: false,
+      styleObject: prefixedPropertyStyleObject('clipPath', this.makePolygonPoints(0, 0))
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.isStatic) return;
+
+    window.addEventListener('scroll', () => {
```

Comment:
This should be debounced.

Reply:
We should also kill the listener when we unmount:
https://stackoverflow.com/a/29726000

Reply:
I wonder though if we could add it to the `onScroll` event of the component?

Reply:
Good spot – in my haste to get it working I stripped out the debounce logic. I'll re-add it (and remove it on unmount).

Not sure I follow re. the `onScroll` even of the component. I think it's always going to have to be the `window` that we listen to.